syntax = "proto3";
package pulse.cdm.bind;
option java_package = "com.kitware.pulse.cdm.bind";
option csharp_namespace = "pulse.cdm.bind";
option optimize_for = SPEED;

import "pulse/cdm/bind/Actions.proto";
import "pulse/cdm/bind/ECMO.proto";

message ECMOActionData
{
  EquipmentActionData                          EquipmentAction             = 1;
}

/** @brief The configuration of the ECMO to use. */
message ECMOConfigurationData
{
 ECMOActionData                                ECMOAction                  = 1;
  oneof Option
  {
    ECMOSettingsData                           Settings                    = 2;/**<< @brief A settings object with properties to set in the system object. */
    string                                     SettingsFile                = 3;/**<< @brief File containing a settings object with properties to set in the system object. */
  }
  eMergeType                                   MergeType                   = 4;/**<< @brief How to apply provided configuration to internal data model. */
}

